Sources/MessageIndex.php

```diff
@@ -326,7 +328,6 @@ function MessageIndex()
 		WHERE t.id_board = {int:current_board} '
 			. (!$modSettings['postmod_active'] || $context['can_approve_posts'] ? '' : '
 			AND (t.approved = {int:is_approved}' . ($user_info['is_guest'] ? '' : ' OR t.id_member_started = {int:current_member}') . ')') . '
-		ORDER BY is_sticky' . ($fake_ascending ? '' : ' DESC') . ', ' . $_REQUEST['sort'] . ($ascending ? '' : ' DESC') . '
```

Comment:
why you throw this out?

Reply:
This line does not affect the script but causes an error when using a hook.
Just test it.

Reply:
what kind of error?

Reply:
since ther is a limit with offset,
the result get changed by this removing.

Reply:
I'm curious about how this line caused an error while using a hook. But at any rate, it is useless since the very same line appears at the end of the main query below anyway.

Reply:
like i said it's not the same,
so need more information what the issue is here.
the var name already mention,
the table should be sorted.

Reply:
Oh, yes, you are right, @albertlast. Since there's a LIMIT and OFFSET in this subquery, an ORDER BY needs to be here.

Reply:
Maybe we can/should remove this one instead?
https://github.com/SimpleMachines/SMF2.1/blob/d25c92908fbbdeb9330f7bc35e92caec58616e11/Sources/MessageIndex.php#L362

Reply:
since we don't know what we try to fix,
it make no sense to guess.

Reply:
No, I don't mean as a fix for whatever error @dragomano has been getting. I mean that it doesn't seem like the second ORDER BY clause is necessary at all, since the subquery has already ordered the results.

Reply:
nope could be changed in the middle of the road.

Reply:
Ok, I have made other changes.

Reply:
> what kind of error?

> Unknown column 'rating' in 'order clause'

I just tried to add another kind of sorting using hooks, using a third-party mod's table and had got errors. The suggested changes eliminate these errors and at the same time do not change the logic of the engine.

Reply:
Why not introduce two new var for the sort part?
I mean in normal case you don't want to add in the sort table column,
because it's affect the performance of the query.

Reply:
Example?

Reply:
well an own set for the sort table query part